tests/unit/db/test_ingest.py

```diff
@@ -26,55 +37,198 @@ def test_cve_data_ingested(self, cve_examples: Path) -> None:
         for cve in nvddb.cve_all():
             assert isinstance(cve, CVE)
 
-    @pytest.mark.parametrize(
-        ("offset", "last_mod_start_date"),
-        [
-            (None, None),
-            (0, None),
-            (42, None),
-            (None, datetime.now(timezone.utc)),
-            (42, datetime.now(timezone.utc)),
-        ],
-    )
-    def test_cve_api_args(self, offset: int | None, last_mod_start_date: datetime | None) -> None:
+    @pytest.mark.parametrize("update", [False, False, False, True, True])
+    def test_cve_api_args(self, update: bool) -> None:
         with patch("bomsquad.vulndb.db.ingest.NVD.vulnerabilities") as vulns:
-            if offset:
-                Ingest.cve(offset, last_mod_start_date=last_mod_start_date)
-            else:
-                Ingest.cve(last_mod_start_date=last_mod_start_date)
+            Ingest.cve(update=update)
             assert vulns.call_count == 1
             args, kwargs = vulns.call_args
-            if offset:
-                assert args[0] == offset
+            assert kwargs["offset"] == 0
+            cp = Checkpoints()
+            if update:
+                assert kwargs["last_mod_start_date"] == cp.last_updated("cve")
             else:
-                assert args[0] == 0
-            assert kwargs["last_mod_start_date"] == last_mod_start_date
+                assert kwargs["last_mod_start_date"] is None
 
     def test_cpe_data_ingested(self, cpe_examples: Path) -> None:
         assert nvddb.cpe_count() == len(list(cpe_examples.iterdir()))
         for cpe in nvddb.cpe_all():
             assert isinstance(cpe, CPE)
 
-    @pytest.mark.parametrize(
-        ("offset", "last_mod_start_date"),
-        [
-            (None, None),
-            (0, None),
-            (42, None),
-            (None, datetime.now(timezone.utc)),
-            (42, datetime.now(timezone.utc)),
-        ],
-    )
-    def test_cpe_api_args(self, offset: int | None, last_mod_start_date: datetime | None) -> None:
+    @pytest.mark.parametrize("update", [False, False, False, True, True])
+    def test_cpe_api_args(self, update: bool) -> None:
         with patch("bomsquad.vulndb.db.ingest.NVD.products") as products:
-            if offset:
-                Ingest.cpe(offset, last_mod_start_date=last_mod_start_date)
-            else:
-                Ingest.cpe(last_mod_start_date=last_mod_start_date)
+            Ingest.cpe(update=update)
             assert products.call_count == 1
             args, kwargs = products.call_args
-            if offset:
-                assert args[0] == offset
+            assert kwargs["offset"] == 0
+            cp = Checkpoints()
+            if update:
+                assert kwargs["last_mod_start_date"] == cp.last_updated("cpe")
             else:
-                assert args[0] == 0
-            assert kwargs["last_mod_start_date"] == last_mod_start_date
+                assert kwargs["last_mod_start_date"] is None
+
+
```

Comment:
Let's move these to tests/unit/client/test_nvd.py and focus them on testing the NVD.vulnerabilities and NVC.products APIs.  Perhaps we can also put the headers down as json files in tests/example/nvd/cpe and tests/example/nvd/cve as we do for the test data generation (and possibly leverage the existing json bodies there as data results).